src/app/test_executive/test_executive.ml

```diff
@@ -44,29 +51,52 @@ let validate_inputs ~logger inputs (test_config : Test_config.t) :
       "This test uses archive nodes.  archive-image argument cannot be absent \
        for this test" ;
     exit 1 )
-  else Deferred.return ()
+  else
+    let (Test_inputs_with_cli_inputs ((module Inputs), _)) =
+      inputs.test_inputs
+    in
+    if
+      String.(test_name inputs.test (module Inputs) = "mock")
+      && Option.is_none inputs.mock_alias
+      && (Option.is_none @@ Sys.getenv "MOCK_NETWORK")
+    then (
+      [%log fatal]
+        "Must provide either --mock command line arg or set MOCK_NETWORK env \
+         var" ;
+      ignore @@ exit 1 ) ;
+    match Inputs.Engine.name with
+    | "abstract" ->
+        if Option.is_none inputs.config_path then (
+          [%log fatal]
+            "Must provide a config file when using the abstract engine" ;
+          exit 1 )
+        else Deferred.unit
+    | _ ->
+        [%log debug]
+          "Config file is only used for the abstract engine. It will be \
+           ignored." ;
+        Deferred.unit
 
 let engines : engine list =
-  [ ("cloud", (module Integration_test_cloud_engine : Intf.Engine.S)) ]
+  [ ("abstract", (module Integration_test_abstract_engine : Intf.Engine.S))
+  ; ("cloud", (module Integration_test_cloud_engine : Intf.Engine.S))
+  ]
 
 let tests : test list =
-  [ ( "peers-reliability"
```

Comment:
Now that the labels are gone from the test list, are we still capable of targeting a specific test, e.g. TestExecutive.execute "peers-reliability"? The TestnetIntegration.dhall  requires this.

Reply:
Good question! Yes. The test names are declared in their respective modules and a function for extracting the name from the first-class test functor and `Inputs` module is provided via `test_name`. In other words, nothing about this functionality has changed.